### PR TITLE
Show next current level review (Fixes #20)

### LINKED ISSUE #20: Show next upcoming review for current level
The upcoming reviews display is really nice and slick, but information specifically about current level items (an alternate graph, a display that just says the time that the next current level item will come up, or ideally a local notification when the next current level item is up for review) would be really useful for people trying to level up quickly.

### PR DESCRIPTION
**Pull request by turtlemaster19**
In order to fix #20, this commit shows the next current level review.

### CODE REVIEW

**Comment by turtlemaster19** [comment]
Oops, forgot to add `CurrentLevelReviewTimeItem.swift` in the Xcode workspace/project. 🤦‍♂

**Comment by turtlemaster19** [comment]
@davidsansome, ready for review.

File: ios/CurrentLevelReviewTimeItem.swift [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        if !assignment.isLessonStage {
+          reviewDates.append(reviewDate)
+        }
+      }
+    }
+
+    // Sort the list of dates
+    reviewDates.sort()
+
+    if let lastReviewDate = reviewDates.last {
```
Comment (zapan) [inline ios/CurrentLevelReviewTimeItem.swift]:
```suggestion
    if let lastReviewDate = reviewDates.first {
```
Shouldn't reviewDates.first be used at this point? 
In my case, next current level review for radicals is in 9 hours, but it says that will be in 21 hours.

IMG_0945

IMG_0946

In any case, thank you very much for this awesome feature!

Comment (turtlemaster19) [reply ios/CurrentLevelReviewTimeItem.swift]: Fixed it, thanks! I'm at level 19.5 right now, so all of my current level items are at Lesson. 😉

**Comment by turtlemaster19** [comment]
@davidsansome, ready for review.

File: ios/CurrentLevelReviewTimeItem.swift (hunk 1 above)
Comment (zapan) [reply ios/CurrentLevelReviewTimeItem.swift]: You're welcome! 😄

**Comment by zapan** [comment]
Since vocabulary reviews are not necessary to level up, I propose to ignore them.
https://github.com/turtlemaster19/tsurukame/pull/4

**Comment by turtlemaster19** [comment]
After looking at #20, I have decided to ignore radicals also, if and when they aren't necessary to level up.

**Comment by turtlemaster19** [comment]
@davidsansome, ready for review.

**Comment by turtlemaster19** [comment]
@davidsansome, ready to merge.

**Comment by turtlemaster19** [comment]
By the way, when there aren't any current level reviews ready, would you prefer something other than the N/A currently used?

**Comment by zapan** [comment]
> By the way, when there aren't any current level reviews ready, would you prefer something other than the N/A currently used? @zapan?

I think that “Not available” label is right.

**Comment by turtlemaster19** [comment]
@davidsansome, ready to merge/review.

File: ios/CurrentLevelReviewTimeItem.swift [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+class CurrentLevelReviewTimeItem: NSObject, TKMModelItem {
+  let services: TKMServices
+  let currentLevelAssignments: [TKMAssignment]
+
+  @objc init(services: TKMServices, currentLevelAssignments: [TKMAssignment]) {
+    self.services = services
+    self.currentLevelAssignments = currentLevelAssignments
+  }
+
+  func createCell() -> TKMModelCell! {
```
[the remark this answers is not in the document]
Comment (turtlemaster19) [reply ios/CurrentLevelReviewTimeItem.swift]: I stole that code from `LevelTimeRemainingItem.swift`. You might want to change it there, too, if you're going to do that.

File: ios/CurrentLevelReviewTimeItem.swift [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    } else {
+      setRemaining(Date.distantFuture)
+    }
+  }
+
+  private func setRemaining(_ finish: Date) {
+    textLabel!.text = "Next level-up review in"
+
+    if finish < Date() {
+      detailTextLabel!.text = "Now"
```
Comment (davidsansome) [inline ios/CurrentLevelReviewTimeItem.swift]: Also, "Next level-up review in Now" doesn't look very grammatically correct.

How about making textLabel.text "Next level-up review", and then having intervalString prepend "in " to the interval.

Comment (turtlemaster19) [reply ios/CurrentLevelReviewTimeItem.swift]: Maybe we can just drop the in all together

File: ios/CurrentLevelReviewTimeItem.swift [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+      }
+      guruDates.append(assignment.guruDate(for: subject))
+    }
+
+    // Sort the list of dates and remove the most distant 10%.
+    guruDates = Array(guruDates.sorted().dropLast(Int(Double(guruDates.count) * 0.1)))
+    let levelKanjiUnlocked = guruDates.last != nil ? (guruDates.last! != Date.distantFuture) : true
+
+    for assignment in item.currentLevelAssignments {
+      if assignment.subjectType == .vocabulary || levelKanjiUnlocked && assignment.subjectType == .radical {
```
[the remark this answers is not in the document]
Comment (turtlemaster19) [reply ios/CurrentLevelReviewTimeItem.swift]: I was using the fact that `&&` has a higher precedence than `||` in order to only ignore radicals if `levelKanjiUnlocked` is true

[the remark this answers is not in the document]
Comment (davidsansome) [reply ios/CurrentLevelReviewTimeItem.swift]: Aah I see, yep parentheses definitely help, thanks :)

**Pull request merged**